src/pip/_internal/operations/prepare.py

```diff
@@ -513,10 +479,15 @@ def prepare_linked_requirement(
             if download_dir:
                 if link.is_existing_dir():
                     logger.info('Link is a directory, ignoring download_dir')
-                elif local_file and not os.path.exists(
-                    os.path.join(download_dir, link.filename)
-                ):
-                    _copy_file(local_file.path, download_dir, link)
+                elif local_file:
+                    download_location = os.path.join(
+                        download_dir, link.filename
+                    )
+                    if not os.path.exists(download_location):
+                        shutil.copy(local_file.path, download_location)
+                        logger.info(
+                            'Saved %s', display_path(download_location)
+                        )
```

Comment:
Would it be possible to refactor this block (starting from `if download_dir:`) somehow? The deep nesting indentation makes me nervous 🥺

Reply:
Later, since when it gets extracted from here it will be easier to do, since we can use e.g. early returns.